tests/E2E Tests/TokenAcquirerTests/TokenAcquirer.cs

```diff
@@ -39,6 +39,7 @@ public TokenAcquirer()
         [Fact]
         public void TokenAcquirerFactoryDoesNotUseAspNetCoreHost()
         {
+            TokenAcquirerFactory.ResetDefaultInstance();
```

Comment:
are these tests stateful?

Reply:
I believe so, because they were failing on things like collections being altered so the enumeration fails or expecting things to be in cache but not finding them. It wasn't a problem before because this was the only test using it up until the